src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/CompilerServices/Utils.LateBinder.vb

```diff
@@ -44,29 +44,92 @@ Namespace Microsoft.VisualBasic.CompilerServices
+        End Function
+
         Friend Shared Function GetResourceString(ByVal ResourceId As vbErrors) As String
-            Dim id as String = "ID" & CStr(ResourceId)
+            Dim id As String = "ID" & CStr(ResourceId)
             Return SR.GetResourceString(id, id)
         End Function
 
-        Friend Shared Function GetResourceString(ByVal resourceKey As String, ByVal ParamArray args() As String) As String
-            Return SR.Format(resourceKey, args)
+        <System.ComponentModel.EditorBrowsable(ComponentModel.EditorBrowsableState.Never)>
+        Friend Shared Function GetResourceString(ByVal ResourceKey As String) As String
+
+            Dim s As String = Nothing
+
+            Try
+                s = SR.GetResourceString(ResourceKey)
+                ' this may be unknown error, so try getting default message
+                If s Is Nothing Then
+                    s = SR.GetResourceString(VBDefaultErrorID)
+                End If
+
+                'if we have found nothing, get a fallback message.
+                If s Is Nothing Then
+                    s = GetFallbackMessage(ResourceKey)
+                End If
+            Catch ex As StackOverflowException
+                Throw ex
+            Catch ex As OutOfMemoryException
+                Throw ex
+            Catch ex As System.Threading.ThreadAbortException
+                Throw ex
+            Catch
+                s = ResourceMsgDefault
+            End Try
+
+            Return s
+        End Function
+
+        ''' <summary>
+        ''' Retrieves a resource string and formats it by replacing placeholders with parameters.
+        ''' </summary>
+        ''' <param name="ResourceKey">The resource string identifier</param>
+        ''' <param name="Args">An array of parameters used to replace placeholders</param>
+        ''' <returns>The resource string if found or an error message string</returns>
+        Public Shared Function GetResourceString(ByVal ResourceKey As String, ByVal ParamArray Args() As String) As String
+
+            System.Diagnostics.Debug.Assert(Not ResourceKey = "", "ResourceKey is missing")
+            System.Diagnostics.Debug.Assert(Not Args Is Nothing, "No Args")
+
+            Dim UnformattedString As String = Nothing
+            Dim FormattedString As String = Nothing
+            Try
+                'Get unformatted string which may have place holders ie "Hello, {0}. How is {1}?"
+                UnformattedString = GetResourceString(ResourceKey)
+
+                'Replace placeholders with items from the passed in array
+                FormattedString = String.Format(System.Threading.Thread.CurrentThread.CurrentCulture, UnformattedString, Args)
+
+                'Rethrow hosting exceptions
+            Catch ex As StackOverflowException
+                Throw ex
+            Catch ex As OutOfMemoryException
+                Throw ex
+            Catch ex As System.Threading.ThreadAbortException
```

Comment:
Nit, CoreCLR never throws ThreadAbortException because Thread.Abort is not supported.

Reply:
It looks like there are a number of `Catch` clauses for `ThreadAbortException`. Logged #1999.